File: backend/app/schemas/audit.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class AuditLogOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int | None = None
    user_name: str | None = None
    action: str
    resource_type: str
    resource_id: int | None = None
    details: dict[str, Any] | None = None
    ip_address: str
    created_at: datetime


class PaginatedAuditLogs(BaseModel):
    items: list[AuditLogOut]
    total: int
    page: int
    page_size: int

File: frontend/src/api/audit.ts
import api from './client'

export interface AuditLogItem {
  id: number
  user_id: number | null
  user_name: string | null
  action: string
  resource_type: string
  resource_id: number | null
  details: Record<string, unknown> | null
  ip_address: string
  created_at: string
}

export interface PaginatedAuditLogs {
  items: AuditLogItem[]
  total: number
  page: number
  page_size: number
}

export interface AuditLogFilters {
  action?: string
  resource_type?: string
  start_date?: string
  end_date?: string
  page?: number
  page_size?: number
}

export const auditApi = {
  list: (params?: AuditLogFilters) =>
    api.get<PaginatedAuditLogs>('/audit-logs', { params }),
}
